config/peer_config.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package config
+
+import (
+	"net/netip"
+
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+type PeerConfig struct {
+	ID string `mapstructure:"id" json:"id"`
+	IP string `mapstructure:"ip" json:"ip"`
+
+	id ids.NodeID
+	ip netip.AddrPort
+}
+
+func (c *PeerConfig) Validate() error {
```

Comment:
Would it be better to have a constructor `newPeerConfig` that does the validation _before_ building the config?

Reply:
We do validate at construction time in the top-level config constructor: https://github.com/ava-labs/awm-relayer/blob/main/relayer/config/viper.go#L21

Reply:
Not the cleanest pattern, but we have to unmarshal the config and validate it in separate steps regardless, it's just a  question of code organization.